src/main/java/com/epam/aidial/core/controller/FileMetadataController.java

```diff
@@ -40,7 +43,9 @@ protected Future<?> handle(ResourceDescription resource, boolean hasWriteAccess)
         }
         return proxy.getVertx().executeBlocking(() -> {
             try {
-                MetadataBase metadata = storage.listMetadata(resource, token, limit, recursive);
+                MetadataBase metadata = resource.isFolder()
+                        ? storage.listMetadata(resource, token, limit, recursive)
```

Comment:
Can you encapsulate it into file service?

Reply:
Unified in the resource service.